tests/abi/reference_binary.c

```diff
@@ -0,0 +1,25 @@
+/*
+ * This file is distributed under the MIT License. See LICENSE.md for details.
+ */
+
+#include <setjmp.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "functions.h"
+
+ABIDEF void *set_return_value_up(void) {
+  return 0;
+}
+
+int main(void) {
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+  assert(!runtime_endianness_check());
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+  assert(runtime_endianness_check());
+#endif
+
+  return 0;
+};
```

Comment:
Can we merge `reference_binary.c` and `runtime_abi_analysis.c`?

Also, in the `abi/` directory we need a `README.md` documenting what's going on.
Feel free to move everything you deem relevant here from the `README.md` in revng.

Reply:
I don't want the extra functions `analysis` introduces to be caught in the cross-fire when parsing the binary. The reference is like it sounds - only the functions we're interested in while `analysis` also contains the way for them to dump stuff and so on.

Reply:
Uhm, what about:

* `reference_binary.c` -> `abi-test-functions-library`
* `runtime_abi_analysis.c` -> `describe-abi-test-functions`

Reply:
I like the way this sounds!

Reply:
Done: https://github.com/revng/revng-qa/pull/4/commits/614d116fd2d54bdcf22308287880bffe89dde361.